src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -1657,6 +1765,27 @@ public static int MethodWithDefaultParameter(int x, int y = 10)
             return x + y;
         }
 
+        public static int MethodWithOneNonObjectParameter(int x)
+        {
+            return x;
+        }
+
+        public static int MethodWithObjectWithFieldsParameter(XAndYFields fields)
+        {
+            return fields.x + fields.y;
+        }
+
+        public static int MethodWithObjectWithPropertiesParameter(XAndYProperties props)
+        {
+            return props.X + props.Y;
+        }
+
+        public static int MethodWithObjectWithProgressParameter(XAndYPropertiesWithProgress props)
+        {
+            props.P.Report(props.X + props.Y);
+            return props.X + props.Y;
+        }
```

Comment:
Where are the tests for methods that take these single parameters plus the (specially treated) `CancellationToken` parameter?

Reply:
Test added in latest update.